lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php

```diff
@@ -61,8 +61,15 @@ public function addTransfer($paymentName, array $transferInformation)
             $transferInformation['creditorName']
         );
         $transfer->setBic($transferInformation['creditorBic']);
-        $transfer->setRemittanceInformation($transferInformation['remittanceInformation']);
-
+        
+        if(array_key_exists('creditorReferenceInformation', $transferInformation) &&
+            strlen($transferInformation['creditorReferenceInformation'])>5 ) {
+                
+            $transfer->setCreditorReferenceInformation($transferInformation['creditorReferenceInformation']); 
+        }else{
```

Comment:
@monofone  they are not mutually-exclusive. In fact the CreditorReferenceInformation sits under the RemittanceInformation element!

The official docs are in [this MDR zipfile](https://www.iso20022.org/sites/default/files/documents/general/ISO20022MDR_PaymentsInitiation_2016_2017.zip). Inside this archive the document _ISO20022_MDRPart2_PaymentsInitiation_2016_2017_v1.pdf_ describes the structure of the whole **pain.001.001.vv** message.

The numbering of the elements in this same document further shows this. For a  reference to the info that shows this:
2.4.2.18        CreditTransferTransactionInformation (<CdtTrfTxInf>) [1..*] -> page: 31-32 
2.4.2.18.25     CreditorReferenceInformation (<CdtrRefInf>) [0..1] -> page: 66
2.4.2.18.25.2.3 RemittanceInformation (<RmtInf>) [0..1] -> page: 47